Code/Framework/AzCore/CMakeLists.txt

```diff
@@ -96,6 +93,18 @@ ly_add_source_properties(
     VALUES ${ALLOW_SETTINGS_REGISTRY_DEVELOPMENT_OVERRIDES_FLAG}
 )
 
+# Add the O3DE_STACK_CAPTURE_DEPTH define only to the cpp files for the following allocators
+# This reduces re-compile time when the cache variable value changes.
+set(O3DE_STACK_CAPTURE_DEPTH_DEFINE $<$<NOT:$<STREQUAL:"${O3DE_STACK_CAPTURE_DEPTH}","">>:O3DE_STACK_CAPTURE_DEPTH=${O3DE_STACK_CAPTURE_DEPTH}>)
+ly_add_source_properties(
+    SOURCES
+        AzCore/Memory/OSAllocator.cpp
+        AzCore/Memory/PoolAllocator.cpp
+        AzCore/Memory/SystemAllocator.cpp
+    PROPERTY COMPILE_DEFINITIONS
+    VALUES ${O3DE_STACK_CAPTURE_DEPTH_DEFINE}
+)
```

Comment:
Instead of adding O3DE_STACK_CAPTURE_DEPTH_DEFINE here, since O3DE_STACK_CAPTURE_DEPTH is already in Configurations.cmake, it seems like this should just add "O3DE_STACK_CAPTURE_DEPTH=${O3DE_STACK_CAPTURE_DEPTH}" to those three files instead of introducing a level of indirection.

Reply:
Actually there is a problem if user if the user overrides the `O3DE_STACK_CAPTURE_DEPTH` to be empty via `-DO3DE_STACK_CAPTURE_DEPTH=`, then it results in a compiler define of `O3DE_STACK_CAPTURE_DEPTH=` which I believe doesn't work everywhere
![image](https://user-images.githubusercontent.com/56135373/219502272-c2a3ae7a-193f-4e9e-8ab5-bcba656c598f.png)